pandas/_libs/tslibs/parsing.pyx

```diff
@@ -386,7 +386,7 @@ cdef parse_datetime_string_with_reso(
         &out_tzoffset, False
     )
     if not string_to_dts_failed:
-        if dts.ps != 0 or out_local:
+        if out_bestunit == NPY_DATETIMEUNIT.NPY_FR_ns or out_local:
```

Comment:
if out_bestunit == NPY_DATETIMEUNIT.NPY_FR_ns is handled here, should it be removed from the dict on L399-L409?

Reply:
No, both blocks go through the dict to get the string representation of the reso.

This is just to force the returned datetime object to be Timestamp, which has a nanosecond attribute.